src/ast/ast_build_reference_map.c

```diff
@@ -17,13 +22,13 @@ static void _AST_MapExpression(AST *ast, const cypher_astnode_t *exp) {
 	// In case of identifier.
 	if(type == CYPHER_AST_IDENTIFIER) {
 		const char *identifier_name = cypher_ast_identifier_get_name(exp);
-		_AST_UpdateRefMap(ast, identifier_name);
+		_AST_UpdateRefMap(ast, identifier_name, REFERENCED_ENTITY_HAS_IDENTIFIER);
```

Comment:
I must be missing something but if a node or an edge has an alias (identifier) it doesn't mean it is referenced
is this the case where `(x), (n {v:x.v})` and we're marking `x` as referenced ?
if so the name`REFERENCED_ENTITY_HAS_IDENTIFIER` isn't clear enough